core_data_modules/data_models/scheme.py

```diff
@@ -34,40 +38,43 @@ def get_code_with_control_code(self, control_code):
 
     def get_code_with_match_value(self, match_value):
         for code in self.codes:
-            if match_value in code.match_values:
+            if code.match_values is not None and match_value in code.match_values:
                 return code
         raise KeyError("Scheme '{}' (id '{}') does not contain a code with match value '{}'".format(self.name, self.scheme_id, match_value))
 
-    @staticmethod
-    def from_firebase_map(data):
-        scheme = Scheme()
-        scheme.scheme_id = validators.validate_string(data["SchemeID"])
-        scheme.name = validators.validate_string(data["Name"])
-        scheme.version = validators.validate_string(data["Version"])
+    @classmethod
+    def from_firebase_map(cls, data):
+        scheme_id = validators.validate_string(data["SchemeID"])
+        name = validators.validate_string(data["Name"])
+        version = validators.validate_string(data["Version"])
 
+        codes = []
         for code_map in data["Codes"]:
             code = Code.from_firebase_map(code_map)
             assert code.code_id not in code_map.keys(), \
                 "Non-unique Code Id found in scheme: {}".format(code.code_id)
-            scheme.codes.append(code)
+            codes.append(code)
 
+        documentation = None
         if "Documentation" in data.keys():
             doc_map = data["Documentation"]
-            scheme.documentation["URI"] = validators.validate_string(doc_map["URI"])
+            documentation = {
+                "URI": validators.validate_string(doc_map["URI"])
+            }
         
-        return scheme
+        return cls(scheme_id, name, version, codes, documentation)
     
     def to_firebase_map(self):
-        ret = {}
+        ret = dict()
```

Comment:
After doing this for a lot of time, I don't think this really makes sense, in this case it's very obvious from context that this is a dictionary not a set.

I suggest that we standardise on using x = {} only for dictionaries, and use the explicit form for handling sets.

Reply:
I don't agree, I think this form is unpythonic. I don't see any risk -

```

Reply:
I'm fine if it's tested. But the lack of tests for this file have caused an afternoon of pain already; this felt like an extra safety net. 

As a (possibly) side note: this is the first piece of untested code we have allowed in Core, and I'm reluctant to ever let that happen again, even if there were claims of informal testing having worked.

Reply:
>I'm fine if it's tested. But the lack of tests for this file have caused an afternoon of pain already; this felt like an extra safety net.

I need to be persuaded that it's ok for code to be more reasonable if it's been tested.

> As a (possibly) side note: this is the first piece of untested code we have allowed in Core, and I'm reluctant to ever let that happen again, even if there were claims of informal testing having worked.

IMHO this is not an appropriate venue for this discussion